### PR TITLE
[dns-sd] add CI tests for SRP server & Advertising Proxy

### PR DESCRIPTION
**Pull request by wgtdkp**
This commit includes basic service registration & advertising function verification. More tests including name conflicts and multiple devices cases will be in future PR.

We changed the `border_routing` test scripts directory to `border_router` as there will be more tests about Thread border router but not only the routing feature.

Depends-On: openthread/ot-br-posix#666

### CODE REVIEW

File: tests/scripts/thread-cert/border_router/test_advertising_proxy.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        client.srp_client_set_host_name('my-host')
+        client.srp_client_set_host_address('2001::1')
+        client.srp_client_add_service('my-service', '_ipps._tcp', 12345)
+        client.srp_client_start(server.get_addrs()[0], client.get_srp_server_port())
+        self.simulator.go(2)
+
+        self.check_host_and_service(server, client)
+
+        #
+        # 2. Discovery the service by the HOST on the ethernet. This makes sure
```
Comment (Irving-cl) [inline tests/scripts/thread-cert/border_router/test_advertising_proxy.py:105]:
```suggestion
        # 2. Discover the service by the HOST on the ethernet. This makes sure
```

File: tests/scripts/thread-cert/border_router/test_advertising_proxy.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        #    when the SRP client removes it.
+        #
+
+        client.srp_client_remove_host()
+        self.simulator.go(2)
+
+        self.assertIsNone(host.discover_mdns_service('my-service', '_ipps._tcp', 'my-host'))
+
+        #
+        # 4. Check if we can discover the mDNS service again when re-register the
```
Comment (Irving-cl) [inline tests/scripts/thread-cert/border_router/test_advertising_proxy.py:122]:
```suggestion
        # 4. Check if we can discover the mDNS service again when re-registering the
```

**Review by Irving-cl — APPROVED** [review]
LGTM 👍

**Review by librasungirl — APPROVED** [review]

File: tests/scripts/thread-cert/backbone/bbr_5_11_01.py [hunk 3]
```diff
@@ -91,6 +91,9 @@ def test(self):
         self.simulator.go(5)
         self.assertEqual('router', self.nodes[ROUTER2].get_state())
 
+        # The OTBR docker enables SRP Server by default, lets explicitly
+        # disable SRP server to avoid Network Data population.
```
Comment (simonlingoogle) [inline tests/scripts/thread-cert/backbone/bbr_5_11_01.py:95]: should we disable SRP server by default in OTBR docker?

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

File: tests/scripts/thread-cert/node.py [hunk 4]
```diff
@@ -221,6 +222,9 @@ def _setup_sysctl(self):
         self.bash(f'sysctl net.ipv6.conf.{self.ETH_DEV}.accept_ra=2')
         self.bash(f'sysctl net.ipv6.conf.{self.ETH_DEV}.accept_ra_rt_info_max_plen=64')
 
+        # Zeroconf complains that there is no enough BUFS to subscribe multicast groups.
```
Comment (jwhui) [inline tests/scripts/thread-cert/node.py:225]:
```suggestion
        # Zeroconf complains that there is not enough BUFS to subscribe multicast groups.
```

File: tests/scripts/thread-cert/backbone/bbr_5_11_01.py (hunk 3 above)
Comment (wgtdkp) [reply tests/scripts/thread-cert/backbone/bbr_5_11_01.py:95]: I think the SRP server should be auto-started in OTBR when the feature is compiled.

Actually, I would expect our test scripts can tolerate additional Service TLV in network data and we should not need the change here.

**Comment by wgtdkp** [comment]
@jwhui More comments on this PR and openthread/ot-br-posix#666?

It seems the codecov is not working, but I believe this PR is going to increase coverage since we are adding CI tests here :)

**Review by jwhui — APPROVED** [review]

**Pull request merged**